subprojects/files/src/main/java/org/gradle/api/internal/file/collections/jdk7/Jdk7DirectoryWalker.java

```diff
@@ -76,14 +80,14 @@ private FileVisitResult checkStopFlag() {
                 }
 
                 @Override
-                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                     FileVisitDetails details = getFileVisitDetails(file, attrs, false);
                     if (isAllowed(details, spec)) {
-                        if (attrs.isSymbolicLink()) {
-                            // when FileVisitOption.FOLLOW_LINKS, we only get here when link couldn't be followed
-                            throw new GradleException(String.format("Could not list contents of '%s'. Couldn't follow symbolic link.", file));
+                        if (attrs.isRegularFile() || attrs.isDirectory()) {
+                            visitor.visitFile(details);
+                        } else {
+                            LOGGER.debug("Could not read file path '{}'.", file);
```

Comment:
I don't understand why we ignore unreadable files when walking directories now. Shouldn't that still fail?

Reply:
had `warn` previously, @lptr explicitly requested `debug` here

Reply:
This PR changed a lot back and forth, I'll reevaluate everything here once https://github.com/gradle/gradle/pull/9743 is merged

Reply:
reverted